pkg/services/sqlstore/migrations/preferences_mig.go

```diff
@@ -46,4 +49,15 @@ func addPreferencesMigrations(mg *Migrator) {
 	mg.AddMigration("Add column week_start in preferences", NewAddColumnMigration(preferencesV2, &Column{
 		Name: "week_start", Type: DB_NVarchar, Length: 10, Nullable: true,
 	}))
+
+	if mg.Cfg != nil && mg.Cfg.IsFeatureToggleEnabled != nil {
+		if mg.Cfg.IsFeatureToggleEnabled(featuremgmt.FlagNewNavigation) {
+			mg.AddMigration("Add column navbar_preferences in preferences", NewAddColumnMigration(preferencesV2, &Column{
+				Name: "navbar_preferences", Type: DB_Text, Nullable: true,
```

Comment:
```suggestion
				Name: "navbar", Type: DB_Text, Nullable: true,
```

Reply:
This works for navbar, but considering using a more generic column that could hold potentially other future settings would be nice so that we don't need to add additional columns for each new setting

Reply:
Maybe `data` or `json_data`?